app/controllers/katello/api/registry/registry_proxies_controller.rb

```diff
@@ -85,6 +84,223 @@ def registry_authorize
       return false
     end
 
+    def handle_blob_push(props = nil)
```

Comment:
It's doing more than handling blobs -- maybe change the name to something like `run_pre_push_checks` ?

Reply:
I ended up splitting this into two before_actions: `blob_push_validation` and `create_container_repo_if_needed`. I think this will also make the lock implementation cleaner. Does that sound ok to you?

Reply:
I'm looking into if I can move the logic for testing if a container is needed into the `before_action` line

Reply:
My point was more that it's also handling pre-checks for manifests, so having "blob" in the name was a little misleading.

Reply:
How does `container_push_prop_validation` sound?

Reply:
This has been changed, closing :)